### ISSUE TITLE
jobs: `TestJobInTxn/normal_success` fails when stressing pkg/jobs

### ISSUE THREAD

**Issue by pbardea** [cockroachdb/cockroach#45217]
Upon running: `make stress PKG=./pkg/jobs TESTFLAGS='-v -logtostderr NONE' TESTFLAGS='-count 500'`, the test `TestJobInTxn/normal_success` fails almost immediately (I believe on the first run) and consistently on master with:
```
=== RUN   TestJobInTxn/normal_success
I200219 18:40:20.174106 40427 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:53450 Attrs: Locality: ServerVersion:19.2-12 BuildTag:v20.1.0-beta.1-187-gbd6dd4122c StartedAt:1582137620083038000 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:53451} ClusterID:e9bc7a1e-8cc0-4664-8912-f69320b7f1ca StartedAt:1582137620083038000 LastUp:1582137620083038000}
I200219 18:40:20.174328 40425 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x67b1440]

goroutine 40690 [running]:
testing.tRunner.func1(0xc00126d200)
	/usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:874 +0x3a3
panic(0x7067fe0, 0x9b14f40)
	/usr/local/Cellar/go/1.13.4/libexec/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).ID(...)
	/Users/pbardea/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:130
github.com/cockroachdb/cockroach/pkg/jobs_test.TestJobInTxn.func6(0xc00126d200)
	/Users/pbardea/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs_test.go:1766 +0x220
testing.tRunner(0xc00126d200, 0xc0032e3dc0)
	/usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:960 +0x350
```

Note, that the test does not flake when stressing the test by itself: `make stress PKG=./pkg/jobs TESTS=TestJobInTxn`

**Comment by spaskob** [comment]
Thanks Paul, this is interesting, if I omit `-count 500` the test passes and I cannot repro:
```
make stress PKG=./pkg/jobs TESTFLAGS='-v -logtostderr NONE'
```
It's bizarre why would adding `TESTFLAGS='-count 500'` cause a change in test behavior?

**Comment by pbardea** [comment]
Hmm, the count flag in tests actually specifies that amount of times to run a given test. Given that information it looked like there was something off with cleaning up the state of the test in the second round. After a bit of poking, I found that the mock plan hooks that were registered were not cleaned up, and this caused issues for this test when it ran it the second time (since the old plan hooks were in place). This should be fixed by something like #45236.
